pkg/cloud-provider/cloudapi/aws/aws_security.go

```diff
@@ -749,12 +775,18 @@ func (c *awsCloud) UpdateSecurityGroupMembers(groupIdentifier *securitygroup.Clo
 	return nil
 }
 
-func (c *awsCloud) DeleteSecurityGroup(groupIdentifier *securitygroup.CloudResourceID, membershipOnly bool) error {
+func (c *awsCloud) DeleteSecurityGroup(groupIdentifier *securitygroup.CloudResource, membershipOnly bool) error {
 	mutex.Lock()
 	defer mutex.Unlock()
 
-	vpcID := groupIdentifier.Vpc
-	accCfg := c.getVpcAccount(vpcID)
+	vpcID := groupIdentifier.Name.Vpc
+	// extract namespacedName from accountID
+	namespacedName, err := extractNamespacedName(groupIdentifier.AccountID)
```

Comment:
the goal is to extract accountcfg. So, I think we should change GetCloudAccountByName to GetCloudAccountByAccountId and extract the key in that function.

Reply:
namespacedName is of type types.NamespacedName. AccountID is a string which is string representation of namespacedName. 
AccountCfgs are stored in a map which are keyed using namespacedName. So, i think it is good to keep the function GetCloudAccountByName as is. As it needs key to be namespacedName, we should convert accountID to types.NamespacedName format and then invoke GetCloudAccountByName. Let me know what are your thoughts.

Reply:
Right now namespacedName is not used in the caller  function apart from adding as input to GetCloudAccountByName. Thats why I am suggesting move fetching namespace, name key inside GetCloudAccountByID.

Reply:
GetCloudAccountByName is called from multiple places and it has namespacedName as input.